# Import des Modules

In [33]:
import pandas as pd
from pathlib import Path


# Import des données pour l'analyse exploratoire

In [34]:
df=pd.read_csv(Path("..") /"data" /"2_processed"/ "paquets_phrases.csv", encoding="utf-8")
df.head()

,nom_fichier,id_paquet,phrases_paquet
0,1893_20_Le_docteur_Pascal._clean.txt,0,Dans la chaleur de l’ardente après-midi de jui...
1,1893_20_Le_docteur_Pascal._clean.txt,1,"Oh! Monsieur, la religion n’a jamais fait de m..."
2,1893_20_Le_docteur_Pascal._clean.txt,2,s’écria la jeune fille. Mais elle était lancée...
3,1893_20_Le_docteur_Pascal._clean.txt,3,"Certes, il n’y a pas de honte à cela; seulemen..."
4,1893_20_Le_docteur_Pascal._clean.txt,4,"On m’attend chez moi, on ne doit plus savoir c..."


# Statistiques descriptives sur le nombre de mots par paquet de phrases

In [35]:
df_phrases_mots = df["phrases_paquet"].str.split().str.len()
df_phrases_mots.describe()

count    2671.000000
mean     1566.187196
std       370.428049
min       145.000000
25%      1317.000000
50%      1520.000000
75%      1769.000000
max      4666.000000
Name: phrases_paquet, dtype: float64

In [36]:
df_phrases_mots

0       1726
1       1594
2       1533
3       1379
4       1761
        ... 
2666    1988
2667    1931
2668    1747
2669    1797
2670    1117
Name: phrases_paquet, Length: 2671, dtype: int64

In [37]:
print(df_phrases_mots[df_phrases_mots < 800].count())
print(df_phrases_mots[df_phrases_mots > 3000].count())
print(df_phrases_mots[df_phrases_mots > 4000].count())


17
6
1


- Nous pouvons constater apres la vérification de la distribution du nombres de mots par paquet :

-premierement, le corpus est composé de 2671 paquets de phrases.

-la moyenne de mots par paquet est de 1566 mots avec une médiane à 1520 mots, ce qui indique que la distribution est légèrement asymétrique vers les valeurs plus élevées mais homogène.

- pour ce qui est des valeurs extrêmes :

 17 / 2671 ≈ 0,64 % c'est négligeable

 6 / 2671 ≈ 0,22 % c'est encore une fois négligeable 

 Nous pouvons les ignorer pour l'analyse ainsi que le pretraitement du corpus. Cependant, il est important de les garder à l'esprit pour les étapes suivantes, notamment lors de la vectorisation et de la modélisation, afin de s'assurer que ces paquets ne biaisent pas les résultats.